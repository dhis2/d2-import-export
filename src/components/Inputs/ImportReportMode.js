import React from 'react'
import i18n from '@dhis2/d2-i18n'
import { importReportModeOptions } from '../../utils/options'
import { RadioGroupField } from '../RadioGroup'

const NAME = 'importReportMode'
const DATATEST = 'input-import-report-mode'
const LABEL = i18n.t('Import report mode')

const ImportReportMode = () => (
    <RadioGroupField
        name={NAME}
        label={LABEL}
        options={importReportModeOptions}
        dataTest={DATATEST}
    />
)

export { ImportReportMode }
