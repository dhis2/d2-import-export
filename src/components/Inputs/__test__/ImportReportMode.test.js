import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { ImportReportMode } from '../ImportReportMode'

jest.mock('react-final-form')

describe('Input component - ImportReportMode', () => {
    it('should render correctly', () => {
        const file = shallow(<ImportReportMode />)

        expect(toJson(file)).toMatchSnapshot()
    })
})
