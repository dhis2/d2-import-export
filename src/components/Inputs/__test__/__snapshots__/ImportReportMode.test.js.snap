// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component - ImportReportMode should render correctly 1`] = `
<Field>
  <Label>
    Import report mode
  </Label>
  <RadioGroup
    dataTest="input-import-report-mode"
    label="Import report mode"
    name="importReportMode"
    options={
      Array [
        Object {
          "label": "Errors",
          "value": "ERRORS",
        },
        Object {
          "label": "Full",
          "value": "FULL",
        },
        Object {
          "label": "Debug",
          "value": "DEBUG",
        },
      ]
    }
  />
</Field>
`;
